### PR TITLE
Switch memQuota's dedup window from int32 to google.protobuf.Duration.

### PR DESCRIPTION
**Pull request by geeknoid**
Also, increase memQuota adapter test coverage to 100%. Covering this last line
of functionality made me realize my rolling window counters would overflow, so
I bumped these to 64 bits.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/247)

### CODE REVIEW

File: adapter/memQuota/config/config.proto [hunk 1]
```diff
@@ -16,9 +16,11 @@ syntax = "proto3";
...
 package adapter.memQuota.config;
 
+import "google/protobuf/duration.proto";
+
 option go_package="config";
 
 message Params {
 	// Minimum number of seconds that deduplication is possible for a given operation.
-	int32 min_deduplication_window_seconds = 1;
+	google.protobuf.Duration min_deduplication_window = 1;
```
Comment (douglas-reid) [inline adapter/memQuota/config/config.proto]: nit: rename as `min_deduplication_duration` to match pending style guide.

File: adapter/memQuota/memQuota.go [hunk 2]
```diff
@@ -96,8 +104,11 @@ func newBuilder() builder {
 func (builder) ValidateConfig(cfg adapter.AspectConfig) (ce *adapter.ConfigErrors) {
 	c := cfg.(*config.Params)
 
-	if c.MinDeduplicationWindowSeconds <= 0 {
-		ce = ce.Appendf("MinDeduplicationWindowSeconds", "deduplication window of %d is invalid, must be >= 0", c.MinDeduplicationWindowSeconds)
+	dedupWindow, err := ptypes.Duration(c.MinDeduplicationWindow)
+	if err != nil {
+		ce = ce.Append("MinDeduplicationWidnow", err)
```
Comment (douglas-reid) [inline adapter/memQuota/memQuota.go]: nit: typo (`Widnow`)

File: adapter/memQuota/config/config.proto (hunk 1 above)
Comment (geeknoid) [inline adapter/memQuota/config/config.proto]: Done

File: adapter/memQuota/memQuota.go (hunk 2 above)
Comment (geeknoid) [inline adapter/memQuota/memQuota.go]: Done

**Comment by geeknoid** [comment]
PTAL.

File: adapter/memQuota/memQuota.go [hunk 3]
```diff
@@ -76,12 +79,17 @@ var keyWorkspacePool = sync.Pool{New: func() interface{} { return &keyWorkspace{
...
+		MinDeduplicationWindow: &duration.Duration{Seconds: 1},
+	}
 )
 
 const (
 	// See the rollingWindow comment for a description of what this is for.
 	ticksPerSecond = 10
+
+	// ns/tick
+	nanosPerTick = 100000000
```
Comment (ZackButcher) [inline adapter/memQuota/memQuota.go]: I'd rather see something more descriptive like `nanosPerTick = int64(time.Nanosecond / time.Second / ticksPerSecond)`.

Comment (geeknoid) [inline adapter/memQuota/memQuota.go]: Done.

**Pull request merged**